### PR TITLE
fix(server): add licenses to `BlobInfo` message

### PR DESCRIPTION
**Pull request by nikpivkin**
## Description

```bash
cat report.json | jq '.Results[]?.Packages[]? | select(.Licenses | length > 0) | {name: .Name, licenses: .Licenses}'
{
  "name": "adduser",
  "licenses": [
    "GPL-2.0"
  ]
}
{
  "name": "apt",
  "licenses": [
    "GPL-2.0"
  ]
}
{
  "name": "base-files",
  "licenses": [
    "GPL-3.0"
  ]
}
{
  "name": "base-passwd",
  "licenses": [
    "GPL-2.0",
    "public-domain"
  ]
}
{
  "name": "bash",
  "licenses": [
    "GPL-3.0"
  ]
}
....
```

## Related issues
- Close https://github.com/aquasecurity/trivy/issues/4996

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: rpc/common/service.proto [hunk 1]
```diff
@@ -212,15 +212,37 @@ message DetectedLicense {
   string          name       = 5;
   float           confidence = 6;
   string          link       = 7;
+}
+
+enum LicenseCategory {
```
Comment (nikpivkin) [inline rpc/common/service.proto:217]: @knqyf263 I have taken the `LicenseCategory` enumeration out of the `DetectedLicense` message. This should not affect backwards compatibility, right?

Comment (knqyf263) [reply rpc/common/service.proto:217]: I hope so, but I'm afraid of breaking something. Can you test with an old trivy server and a new trivy client?

Comment (nikpivkin) [reply rpc/common/service.proto:217]: Tested with Trivy v46.0 server and the scan completed successfully.

Comment (knqyf263) [reply rpc/common/service.proto:217]: Thanks!

File: rpc/common/service.proto [hunk 2]
```diff
@@ -212,15 +212,37 @@ message DetectedLicense {
...
+  LICENSE_CATEGORY_UNSPECIFIED  = 0;
```
Comment (knqyf263) [inline rpc/common/service.proto:218]: The generated constants are now like `LicenseCategory_LICENSE_CATEGORY_UNSPECIFIED`. Do we need `LICENSE_CATEGORY_` prefix for these enums?

Comment (nikpivkin) [reply rpc/common/service.proto:218]: @knqyf263 A conflict can happen within a single proto file without using the enum name prefix. https://github.com/protocolbuffers/protobuf/issues/5425

Also protobuf [recommends](https://protobuf.dev/programming-guides/style/#enums) using prefix for enum names.
There is an open [issue](https://github.com/golang/protobuf/issues/513) to remove the prefix in the enum name.

Comment (knqyf263) [reply rpc/common/service.proto:218]: As people are complaining in the issue, it looks really weird.

>I found the following https://github.com/golang/protobuf/issues/513#issuecomment-1672772801 to improve the readability of enumerations:

This is not ideal, but we can accept it. @nikpivkin Can you please rewrite the enum in the way and write a comment on why we need this workaround?

Comment (nikpivkin) [reply rpc/common/service.proto]: Done https://github.com/aquasecurity/trivy/pull/5382/commits/2096c91d71aa4349c38593e6d7a7a109d71057d3

**Review by knqyf263 — APPROVED** [review]

File: rpc/common/service.proto (hunk 2 above)
Comment (knqyf263) [reply rpc/common/service.proto]: Thanks!

**Pull request merged**